processing/src/main/java/org/apache/druid/frame/channel/ReadableInputStreamFrameChannel.java

```diff
@@ -150,6 +152,10 @@ public void close()
 
   private void startReading()
   {
+    if (!toStart) {
```

Comment:
Is this change required, once the cause is fixed in the `S3StorageConnector`? (If so, then can you document the behavior that we buffer the input stream lazily)